tiledb/sm/query/writer.cc

```diff
@@ -1494,7 +1428,14 @@ Status Writer::init_tile_dense_cell_range_iters(
   domain->get_tile_domain(&subarray[0], &tile_domain[0]);
   for (unsigned i = 0; i < dim_num; ++i)
     tile_coords[i] = tile_domain[2 * i];
-  auto tile_num = domain->tile_num<T>(&subarray[0]);
+
+  // TODO: fix
```

Comment:
What's wrong with this? Could you expand on the comment?

Reply:
This goal of this PR (and subsequent PRs) is to remove templates and assume heterogenous multi-dimensional ranges. Gradually every range will become an `NDRange` object.